src/arithmetic.jl

```diff
@@ -41,4 +41,10 @@ function broadcasted(::typeof(+), r::StepRange{ZonedDateTime}, p::DatePeriod)
     return StepRange(start, step, stop)
 end
 
-broadcasted(::typeof(-), r::StepRange{ZonedDateTime}, p::DatePeriod) = broadcast(+, r, -p)
+function broadcasted(::typeof(+), r::StepRange{ZonedDateTime}, p::TimePeriod)
+    StepRange(r.start + p, r.step, r.stop + p)
+end
+
+broadcasted(::typeof(+), p::Period, r::StepRange{ZonedDateTime}) = broadcasted(+, r, p)
+broadcasted(::typeof(-), r::StepRange{ZonedDateTime}, p::Period) = broadcasted(+, r, -p)
```

Comment:
Curious about why the change from `DatePeriod` to `Period`

Reply:
Good question. Really I updated the code from:
```julia
broadcasted(::typeof(+), r::StepRange{ZonedDateTime}, p::DatePeriod) = ...
broadcasted(::typeof(-), r::StepRange{ZonedDateTime}, p::DatePeriod) = broadcasted(+, r, -p)
```
to
```julia
broadcasted(::typeof(+), r::StepRange{ZonedDateTime}, p::DatePeriod) = ...
broadcasted(::typeof(+), p::DatePeriod, r::StepRange{ZonedDateTime}) = broadcasted(+, r, p)
broadcasted(::typeof(-), r::StepRange{ZonedDateTime}, p::DatePeriod) = broadcasted(+, r, -p)
broadcasted(::typeof(-), p::DatePeriod, r::StepRange{ZonedDateTime}) = broadcasted(-, r, p)

broadcasted(::typeof(+), r::StepRange{ZonedDateTime}, p::TimePeriod) = ...
broadcasted(::typeof(+), p::TimePeriod, r::StepRange{ZonedDateTime}) = broadcasted(+, r, p)
broadcasted(::typeof(-), r::StepRange{ZonedDateTime}, p::TimePeriod) = broadcasted(+, r, -p)
broadcasted(::typeof(-), p::TimePeriod, r::StepRange{ZonedDateTime}) = broadcasted(-, r, p)
```
However, since `DatePeriod`/`TimePeriod` are subtypes of `Period` I can use a more generic signature to avoid duplicating methods. Doing that results in what's currently coded:
```julia
broadcasted(::typeof(+), r::StepRange{ZonedDateTime}, p::DatePeriod) = ...
broadcasted(::typeof(+), r::StepRange{ZonedDateTime}, p::TimePeriod) = ...

broadcasted(::typeof(+), p::Period, r::StepRange{ZonedDateTime}) = broadcasted(+, r, p)
broadcasted(::typeof(-), r::StepRange{ZonedDateTime}, p::Period) = broadcasted(+, r, -p)
broadcasted(::typeof(-), p::Period, r::StepRange{ZonedDateTime}) = broadcasted(-, r, p)
```